packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -54,6 +54,14 @@ abstract class SerializationManager {
   T deserialize<T>(dynamic data, [Type? t]) {
     t ??= T;
 
+    if (data == null) {
+      if (null is! T) {
+        throw FormatException(
+          'Failed to deserialize data. The provided data is null, and the target type $T cannot be instantiated.',
+        );
+      }
+    }
```

Comment:
Question: I'm not quite sure I follow why this is required after this change?
Do we have a test that validates the need of this scenario to make it easer to understand?